web/src/test/auto/headless/engine/keyboard/keyboard-loading.tests.ts

```diff
@@ -0,0 +1,132 @@
+import { assert } from 'chai';
+
+import { createRequire } from 'module';
+const require = createRequire(import.meta.url);
+
+import { KeyboardHarness, MinimalKeymanGlobal, KeyboardDownloadError, DeviceSpec, KeyboardMissingError } from 'keyman/engine/keyboard';
+import { KeyboardInterface, Mock } from 'keyman/engine/js-processor';
+import { NodeKeyboardLoader } from 'keyman/engine/keyboard/node-keyboard-loader';
+
+// async function assertThrowsAsync(fn: () => Promise<any>, message?: string): Promise<void>;
+async function assertThrowsAsync(fn: () => Promise<any>, type?: any, message?: string): Promise<void> {
+  if (typeof (type) === 'string') {
+    message = type;
+    type = undefined;
+  }
+  try {
+    await fn();
+    assert.fail('Expected function to throw an error, but it did not.');
+  } catch (err) {
+    if (type) {
+      assert.isTrue(err instanceof type, `Expected error to be of type ${type.name}, but got ${err.constructor.name}`);
+    }
+    if (message) {
+      assert.equal((err as Error).message, message);
+    }
+  }
+}
+
+function assertThrows(fn: () => any, message?: string): void;
+function assertThrows(fn: () => any, type?: any, message?: string): void {
+  if (typeof (type) === 'string') {
+    message = type;
+    type = undefined;
+  }
+  assert.throws(fn, type, message);
+}
+
+describe('Headless keyboard loading', function() {
```

Comment:
We are going to need tests for in-browser keyboard loading too I guess

Reply:
done